### ISSUE TITLE
Re-enable automatic API documentation generators in CI

### ISSUE THREAD

**Issue by CPTNB** [opensearch-project/OpenSearch-Dashboards#2032]
The codebase has references to APIExtractor but those scripts aren't used.  This issue is to turn that back on so that we can get JSDoc/APIExtractor artifacts created as part of CI.  Actually hosting or putting this documentation anywhere is out of scope and handled in another ticket.

**Comment by CPTNB** [comment]
Re-enabling api-extractor is blocked by [TS compilation #1660 ](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1660).

Investigating JSDoc to get something in the meantime
